src/meshapi/views/forms.py

```diff
@@ -395,29 +398,40 @@ def network_number_assignment(request: Request) -> Response:
         logging.exception(f'NN Request failed. Could not get Install w/ Install Number "{r.install_number}"')
         return Response({"detail": "Install Number not found"}, status=status.HTTP_404_NOT_FOUND)
 
-    # Check if the install already has a network number
-    if nn_install.node is not None:
-        message = f"This Install Number ({r.install_number}) already has a "
-        f"Network Number ({nn_install.node.network_number}) associated with it!"
-        logging.warn(message)
-        return Response(
-            {
-                "detail": message,
-                "building_id": nn_install.building.id,
-                "install_number": nn_install.install_number,
-                "network_number": nn_install.node.network_number,
-                "created": False,
-            },
-            status=status.HTTP_200_OK,
-        )
+    # Track if we have made any changes, so we know what status code to return
+    dirty = False
```

Comment:
I wonder if it would be possible to use django's `.original` or something at the end to check if anything changed, and just always try to save. Would that be simpler than a dirty bit?

Reply:
We could do this, but it would require an extra DB call for each model we are saving, to fetch the pre-edited copy. Django does not keep track of the original on our behalf unfortunately.

I don't think those extra DB calls are that bad from a performance perspective (we're doing a ton of DB calls here anyway). But I think they represent a step up in complexity from the dirty bit approach. Especially when you consider the edge cases around the objects not necessarily existing, etc. 

If you feel strongly we could do it, there might be a robustness argument for it